core/txpool/txpool_test.go

```diff
@@ -158,20 +158,23 @@ func dynamicFeeTx(nonce uint64, gaslimit uint64, gasFee *big.Int, tip *big.Int,
 	return tx
 }
 
-func setupPool() (*TxPool, *ecdsa.PrivateKey) {
+func setupPool() (*TxPool, *ecdsa.PrivateKey, error) {
 	return setupPoolWithConfig(params.TestChainConfig)
 }
 
-func setupPoolWithConfig(config *params.ChainConfig) (*TxPool, *ecdsa.PrivateKey) {
+func setupPoolWithConfig(config *params.ChainConfig) (*TxPool, *ecdsa.PrivateKey, error) {
 	statedb, _ := state.New(common.Hash{}, state.NewDatabase(rawdb.NewMemoryDatabase()), nil)
 	blockchain := newTestBlockchain(statedb, 10000000, new(event.Feed))
 
 	key, _ := crypto.GenerateKey()
-	pool := NewTxPool(testTxPoolConfig, config, blockchain)
+	pool, err := NewTxPool(testTxPoolConfig, config, blockchain)
+	if err != nil {
+		return nil, nil, err
+	}
 
 	// wait for the pool to initialize
 	<-pool.initDoneCh
-	return pool, key
+	return pool, key, nil
 }
```

Comment:
it looks like neither of these should ever error in tests and we add `t.Fatal(...)` everyplace that it does. Could we change this to panic if there's an unexpected error and reduce the diff of this file?

Reply:
Yeah we can panic instead